### ISSUE TITLE
Add ability to rename groups in the groups editor

### ISSUE THREAD

**Issue by Ortham** [loot/loot#1789]
As suggested by @IllusiveMan196 and first mentioned in #1750.

Group names are read-only, so this would be done by creating a new group with the new name and same description and load after data, and then removing the group with the old name.

**Comment by Ortham** [comment]
I was thinking about this earlier, and:

* this would have to be restricted to user-created groups, as groups are identified by their name, so having a different name for a group than what's in the masterlist would cause problems as the masterlists are updated or the user installs new plugins
* when applying this change:
  * the group with the old name will need to be removed
  * a new group with the new name and the same description and load after metadata as the old group will need to be created
  * any "load after" references to the old group in other groups' metadata will need to be replaced
  * any plugins that were in the old group will need their metadata updated to put them in the new group

This also interacts with #1790: if a plugin is added to a group and the group then renamed before exiting the groups editor, the plugin must be added to the group with the new name.

**Comment by sibir-ine** [comment]
Not exactly sure how to improve it either. However, one minor improvement for the current configuration would be to make the drop-down menu & **Add plugin to group** button the same width as the divider line & the **Auto arrange groups** button.

**Comment by Ortham** [comment]
> However, one minor improvement for the current configuration would be to make the drop-down menu & **Add plugin to group** button the same width as the divider line & the **Auto arrange groups** button.

Good catch, I've fixed that and rearranged things slightly to separate the auto arrange button from the others:

![image](https://user-images.githubusercontent.com/2430738/205256691-f3d2aeed-91bc-4920-9b40-77569f4589d3.png)

**Comment by sibir-ine** [comment]
That's better, the extra divider helps keep unrelated functions (at least from the user's perspective) separated. Besides the UI, something else that would be beneficial is the ability to remove plugins from groups within the editor. Hopefully it's not too much more complicated to implement now that you've added the ability to add plugins. Besides its inherent utility, I can imagine it being a bit cumbersome for users to add a plugin in the groups editor & then need to exit it if they change their mind.

@IllusiveMan196 also requested the additional ability to type in plugins to add which I imagine would be quite useful for larger load orders. I noticed that with the drop-down focused, you can type to navigate to an item, but it can be a bit finicky. Maybe Qt offers some kind of drop-down menu that allows you filter items down with a text field? They also requested the ability to drag plugins from the main window like you can with the metadata editor, but I imagine that'd be a much more difficult task to accomplish & probably not worth it with their other request.

**Comment by sibir-ine** [comment]
Yeah, I think filtering would be enough. Re removing, you're right, I hadn't considered that it's more complicated than it seems. Perhaps just reverting changes made to a plugin within a session of the groups editor would be sufficient without opening that can of worms.

**Comment by Ortham** [comment]
@sibir-ine The latest build on the `improve-groups-editor` branch now has autocomplete, can you try it out?

**Comment by sibir-ine** [comment]
Works really well 👍 @IllusiveMan196, would you like to try it out as well to see if it operates how you imagined it would?

**Comment by IllusiveMan196** [comment]
This works as intended and a bit better even. Thanks. "Group auto-arrange" button produced bit broken-ish version of my current graph, but that is not quite an issue really.

**Comment by Ortham** [comment]
Great, done as of 9487c7cb2e478f4de4733d26b9cfc6f74c1e80a4.
